src/test/ui/catch-block-type-error.stderr

```diff
@@ -0,0 +1,25 @@
+error[E0271]: type mismatch resolving `<std::option::Option<f32> as std::ops::Try>::Ok == {integer}`
+  --> $DIR/catch-block-type-error.rs:18:9
+   |
+LL |         42
+   |         ^^ expected f32, found integral variable
+   |
+   = note: expected type `f32`
+              found type `{integer}`
+
+error[E0271]: type mismatch resolving `<std::option::Option<i32> as std::ops::Try>::Ok == ()`
+  --> $DIR/catch-block-type-error.rs:22:35
+   |
+LL |       let _: Option<i32> = do catch {
+   |  ___________________________________^
+LL | |         //~^ ERROR type mismatch
+LL | |         foo()?;
+LL | |     };
+   | |_____^ expected i32, found ()
```

Comment:
Hmm, this is ungreat. Not sure what span would be best, maybe @estebank has thoughts. I think maybe just pointing to the end-brace would be good?:

```
error[E0271]: type mismatch resolving `<std::option::Option<i32> as std::ops::Try>::Ok == ()`
  --> $DIR/catch-block-type-error.rs:22:35
   |
LL |       };
   |       ^ expected i32, found ()
   |
   = note: expected type `i32`
              found type `()`
```

Also, the "type mismatch" line is kind of a mess, but that's separate I guess and maybe hard to fix -- we might be able to use the "extra info" from the span to help.

Reply:
That seems plausible, though note that the multi-line span happens today for blocks:
```rust
error[E0308]: mismatched types
 --> src/main.rs:2:19
  |
2 |       let x: i32  = {
  |  ___________________^
3 | |         println!("asdf");
4 | |         3;
5 | |         println!("asdf");
6 | |     };
  | |_____^ expected i32, found ()
  |
  = note: expected type `i32`
             found type `()`
```

I haven't looked into span futzing yet, so I don't know my unknowns.  A few quick pointers would be great, or if you wanted to use me as a tester for a rustc book page...

Reply:
> note that the multi-line span happens today for blocks

yes, but that doesn't make it good =)

> I haven't looked into span futzing yet, so I don't know my unknowns

I believe that [the `end_point` function is what you want](https://doc.rust-lang.org/nightly/nightly-rustc/syntax/codemap/struct.CodeMap.html#method.end_point)

Reply:
Thanks!  That did the trick.  And huzzah for hosted compiler docs :tada: